tests/playwright-test/global-setup.spec.ts

```diff
@@ -386,3 +386,43 @@ test('teardown after error', async ({ runInlineTest }) => {
     'teardown 1',
   ]);
 });
+
+test('globalSetup should support multiple', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'playwright.config.ts': `
+      module.exports = {
+        globalSetup: ['./globalSetup1.ts','./globalSetup2.ts','./globalSetup3.ts','./globalSetup4.ts'],
+        globalTeardown: ['./globalTeardown1.ts', './globalTeardown2.ts'],
+      };
+    `,
+    'globalSetup1.ts': `module.exports = () => { console.log('%%globalSetup1'); return () => { console.log('%%globalSetup1Function'); throw new Error('kaboom'); } };`,
+    'globalSetup2.ts': `module.exports = () => console.log('%%globalSetup2');`,
+    'globalSetup3.ts': `module.exports = () => { console.log('%%globalSetup3'); return () => console.log('%%globalSetup3Function'); }`,
+    'globalSetup4.ts': `module.exports = () => console.log('%%globalSetup4');`,
+    'globalTeardown1.ts': `module.exports = () => console.log('%%globalTeardown1')`,
+    'globalTeardown2.ts': `module.exports = () => console.log('%%globalTeardown2');`,
```

Comment:
I'd also throw here, and check that it did not prevent globalTeardown1 from running.

Reply:
Did that in https://github.com/microsoft/playwright/pull/32955/commits/a72ec6e8245a1725995400429a4ed26481c7cc9b. This behaviour is surprising to me, was somehow expecting a failing `teardown` to stop further teardowns from being executed.

With that in mind, I don't think that a teardown function returned from `globalSetup` should behave any differently than a `globalTeardown` when it throws an error. I've made sure that a failing function doesn't block any `globalTeardown` script from being executed in https://github.com/microsoft/playwright/pull/32955/commits/9b785662f6e04f13f183985cf1f3b341e3480db3. Would be lovely if you could give that another look.

I think this could benefit from another refactoring where we pull `globalSetup` and `globalTeardown` scripts into separate tasks, but i'd like to get this PR merged first and then refactor in a follow-up.

Reply:
On a second note, this could be considered a breaking change. If folks are currently relying on the `globalSetup` result throwing an error to prevent `globalTeardown` from being executed, that's broken with https://github.com/microsoft/playwright/commit/9b785662f6e04f13f183985cf1f3b341e3480db3. So maybe we should just stick with the incongruence.

Reply:
I agree, let's not touch the relative behavior of `globalSetup` returned function vs `globalTeardown`. The aim here is to support multiple, and our general rule is that all teardowns run even if one of them fails. This applies to fixtures, afterEach and afterAll hooks, and now to global teardown as well.